proposals/0000-dependent-type-design.rst

```diff
@@ -0,0 +1,388 @@
+in evaluating other proposals.
+
+Examples
+--------
+* If this current proposal is accepted, I would expect the committee to accept
+  `#291`_ (with significant revisions to the text, but not the spirit) and
+  `#270`_ (perhaps with significant revisions to the details). `#281`_ could
+  then be drastically simplified and designed to work only in the subset of
+  the language that contains no puns; my hope is then that `#281`_, too, would
+  be accepted.
+
+* This proposal frequently talks about *ergonomic* dependent types, but it does
+  not define those terms. I will not attempt to define them here, because what
+  counts as ergonomic is generally a matter of taste. (What counts as dependent
+  types is also surprisingly flexible; some have claimed that Haskell already
+  has dependent types.) However, one key component of ergonomics for dependent
+  types is (for me) that users do not need to think about a syntactic difference
+  between types and terms.
+
+  Haskell will always have (I sure hope) a separation
+  between compile-time computation and runtime computation, and we will always
+  (I sure hope) erase irrelevant compile-time terms. (More simply: Haskell has
+  and will keep *type erasure*. It's just that the word "type" is becoming harder
+  to pin down, so I have to say "compile-time terms".) Yet my hope for dependent
+  types in Haskell should not require a syntactic distinction between types and
+  terms. While such a distinction is not wholly incompatible with dependency,
+  I think it would add significant complication and cognitive overhead to understanding
+  dependent types in Haskell.
+
+Effect and Interactions
+-----------------------
+* By accepting this proposal, the committee reaffirms Haskell's status as
+  an evolving, forward-thinking language, excited to adopt new ideas.
+
+* This proposal does *not* invalidate any current syntax, nor does it mean
+  that GHC will not consider non-dependent-type proposals. This proposal is
+  all about informing judgment calls, mostly around concrete syntax, in other
+  proposals.
+
+* This proposal does *not* eliminate criterion 1. It simply makes explicit that
+  we care (deeply) about criterion 2. At all times, we would continue to try
+  to meet both criteria.
+  
+* A rejection of this proposal would likely lead to some "brain drain": I am
+  aware of a number of active contributors to our community who are excited
+  about the possibility of dependent types. Rejecting this proposal may signal
+  to them that Haskell is not interested in what they have to offer; they may
+  join other language communities.
+
+* This proposal does *not* say anything about *backward* compatibility. Specifically,
+  it does not propose that we sacrifice backward compatibility in the service
+  of forward compatibility. It is every expectation that proposals building
+  dependently typed features would maintain backward compatibility. Where that
```

Comment:
I find this very surprising. In my mind one of the biggest draws of Dependent Haskell has been having a better story around type-level programming. As many people have pointed out, the current state of type-level programming in Haskell is often complicated and painful, and could be **simplified** by adding dependent types. Once we have the simpler story, why wouldn't we (slowly) remove old features that have been subsumed?

Reply:
Good point. I've added some text to the proposal on this point, to be uploaded when I'm done with this sweep of comments.